tests/test_tags.py

```diff
@@ -154,3 +157,146 @@ def test_empty_response(self):
         copy_tags = []
         final_tags = coalesce_copy_user_tags(resource, copy_tags, user_tags)
         self.assertEqual(final_tags, [])
+
+
+class CopyRelatedResourceTag(BaseTest):
+    def test_copy_related_resource_tag_all(self):
+        session_factory = self.replay_flight_data("test_tags_copy_related_resource_tags_all")
+        p = self.load_policy(
+            {
+                "name": "copy-related-resource-tags-snapshots-volumes",
+                "resource": "ebs-snapshot",
+                "filters": [
+                    {
+                        "Tags": "empty"
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "copy-related-resource-tag",
+                        "resource": "ebs",
+                        "key": "VolumeId",
+                        "copy_tags": ["*"]
+                    }
+                ]
+            },
+            session_factory=session_factory
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        client = session_factory().client('ec2', 'us-east-1')
+
+        snap = client.describe_snapshots(SnapshotIds=[resources[0]['SnapshotId']])['Snapshots']
+        vol = client.describe_volumes(VolumeIds=[resources[0]['VolumeId']])['Volumes']
+
+        self.assertEqual(len(snap), 1)
+        self.assertEqual(len(vol), 1)
+
+        self.assertEqual(snap[0]['Tags'], vol[0]['Tags'])
+
+    def test_copy_related_resource_tag_partial(self):
+        session_factory = self.replay_flight_data("test_tags_copy_related_resource_tag_partial")
+        p = self.load_policy(
+            {
+                "name": "copy-related-resource-tags-snapshots-volumes",
+                "resource": "ebs-snapshot",
+                "filters": [
+                    {
+                        "Tags": "empty"
+                    }
+                ],
+                "actions": [
+                    {
+                        "type": "copy-related-resource-tag",
+                        "resource": "ebs",
+                        "key": "VolumeId",
+                        "copy_tags": [
+                            "tag1",
+                            "tag3"
+                        ]
+                    }
+                ]
+            },
+            session_factory=session_factory
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        client = session_factory().client('ec2', 'us-east-1')
+
+        snap = client.describe_snapshots(SnapshotIds=[resources[0]['SnapshotId']])['Snapshots']
+        vol = client.describe_volumes(VolumeIds=[resources[0]['VolumeId']])['Volumes']
+
+        self.assertEqual(len(snap), 1)
+        self.assertEqual(len(vol), 1)
```

Comment:
also superflous, ie. the api will return a single resource if its being passed a single resource to filter isn't meaningful to assert.